zeebe/gateway-rest/src/main/java/io/camunda/zeebe/gateway/rest/SearchQueryRequestMapper.java

```diff
@@ -114,27 +113,17 @@ Either<ProblemDetail, T> buildSearchQuery(
   private static Either<List<String>, ProcessInstanceFilter> toProcessInstanceFilter(
       final ProcessInstanceFilterRequest filter) {
     final var builder = FilterBuilders.processInstance();
-    final List<String> validationErrors = new ArrayList<>();
 
     if (filter != null) {
-      final var variableFiltersResult = toVariableValueFilter(filter.getVariables());
-      if (variableFiltersResult.isLeft()) {
-        validationErrors.addAll(variableFiltersResult.getLeft());
-      } else {
-        final var variableFilters = variableFiltersResult.get();
-        if (variableFilters != null) {
-          builder.variable(variableFilters);
-        }
+      final var variableFilters = toVariableValueFilter(filter.getVariables()).get();
```

Comment:
🔧 If there is no expected error result (Left), then simply don't use an Either. It makes the code harder to understand when an Either is involved but never used as such.